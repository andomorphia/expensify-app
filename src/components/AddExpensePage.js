import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';

export class AddExpensePage extends Component {
  static propTypes = {
    startAddExpense: PropTypes.func.isRequired,
    history: PropTypes.object.isRequired,
  }

  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.history.push('/');
  };

  render() {
    return (
      <Fragment>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            onSubmit={this.onSubmit}
          />
        </div>
      </Fragment>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startAddExpense: expense => dispatch(startAddExpense(expense)),
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
