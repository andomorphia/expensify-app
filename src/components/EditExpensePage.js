import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends Component {
  static propTypes = {
    startEditExpense: PropTypes.func,
    startRemoveExpense: PropTypes.func,
    expense: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
  }

  static defaultProps = {
    startEditExpense: undefined,
    startRemoveExpense: undefined,
  }

  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };

  onRemove = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };

  render() {
    return (
      <Fragment>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button
            className="button button--secondary"
            onClick={this.onRemove}
          >
            Remove Expense
          </button>
        </div>
      </Fragment>
    );
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(expense => expense.id === props.match.params.id),
});

const mapDispatchToProps = dispatch => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: id => dispatch(startRemoveExpense(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
