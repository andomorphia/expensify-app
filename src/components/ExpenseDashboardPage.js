import React from 'react';

import ConnectedExpensesSummary from './ExpensesSummary';
import ConnectedExpenseListFilters from './ExpenseListFilters';
import ConnectedExpenseList from './ExpenseList';

const ExpenseDashboardPage = () => (
  <div>
    <ConnectedExpensesSummary />
    <ConnectedExpenseListFilters />
    <ConnectedExpenseList />
  </div>
);

export default ExpenseDashboardPage;
