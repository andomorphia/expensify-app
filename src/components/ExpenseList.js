import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = props => (
  <div>
    {
      props.expenses.length === 0 ? (
        <p>No expenses</p>
      ) : (
        props.expenses.map(expense => (
          <ExpenseListItem key={expense.id} {...expense} />
        ))
      )
    }
  </div>
);

ExpenseList.propTypes = {
  expenses: PropTypes.array,
};

ExpenseList.defaultProps = {
  expenses: [],
};

const mapStateToProps = state => ({
  expenses: selectExpenses(state.expenses, state.filters),
});

export default connect(mapStateToProps)(ExpenseList);
