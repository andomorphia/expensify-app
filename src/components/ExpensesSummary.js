import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import numeral from 'numeral';

import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedExpenseCount = numeral(expenseCount).format('0,0');
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');

  return (
    <h2>Viewing {formattedExpenseCount} {expenseWord} totalling {formattedExpensesTotal}</h2>
  );
};

ExpensesSummary.propTypes = {
  expenseCount: PropTypes.number,
  expensesTotal: PropTypes.number,
};

ExpensesSummary.defaultProps = {
  expenseCount: undefined,
  expensesTotal: undefined,
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
