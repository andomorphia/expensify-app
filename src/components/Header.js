import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';

import { startLogout } from '../actions/auth';

export const Header = ({ startLogoutProcess }) => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/dashboard" activeClassName="is-active">Dashboard</NavLink>
    <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
    <button onClick={startLogoutProcess}>Logout</button>
  </header>
);

Header.propTypes = {
  startLogoutProcess: PropTypes.func,
};

Header.defaultProps = {
  startLogoutProcess: undefined,
};

const mapDispatchToProps = dispatch => ({
  startLogoutProcess: () => dispatch(startLogout()),
});

export default connect(undefined, mapDispatchToProps)(Header);
