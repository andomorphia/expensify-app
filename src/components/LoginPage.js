import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLoginProcess }) => (
  <button onClick={startLoginProcess}>Login</button>
);

LoginPage.propTypes = {
  startLoginProcess: PropTypes.func,
};

LoginPage.defaultProps = {
  startLoginProcess: undefined,
};

const mapDispatchToProps = dispatch => ({
  startLoginProcess: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
