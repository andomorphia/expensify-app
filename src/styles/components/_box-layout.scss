.box-layout {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100vw;
  height: 100vh;
  background: url('/images/bg.jpg');
  background-size: cover;
}

.box-layout__box {
  padding: $l-size $m-size;
  width: 25rem;
  text-align: center;
  background: fade-out(white, .15);
  border-radius: 3px;
}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
}
