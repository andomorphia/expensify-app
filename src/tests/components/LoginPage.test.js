import React from 'react';
import { shallow } from 'enzyme';

import { LoginPage } from '../../components/LoginPage';

describe('LoginPage component rendering: ', () => {
  test('should render LoginPage correctly', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('LoginPage component (startLoginProcess): ', () => {
  test('should handle startLoginProcess', () => {
    const startLoginProcess = jest.fn();
    const wrapper = shallow(<LoginPage startLoginProcess={startLoginProcess} />);
    wrapper.find('button').simulate('click');
    expect(startLoginProcess).toHaveBeenCalled();
  });
});
