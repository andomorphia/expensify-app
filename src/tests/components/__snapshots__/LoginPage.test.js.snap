// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component rendering:  should render LoginPage correctly 1`] = `
<button>
  Login
</button>
`;
